x/group/keeper/msg_server.go

```diff
@@ -485,6 +485,67 @@ func (k Keeper) CreateProposal(goCtx context.Context, req *group.MsgCreatePropos
 	return &group.MsgCreateProposalResponse{ProposalId: id}, nil
 }
 
+func (k Keeper) WithdrawProposal(goCtx context.Context, req *group.MsgWithdrawProposal) (*group.MsgWithdrawProposalResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	id := req.ProposalId
+	admin := req.Admin
+
+	proposal, err := k.getProposal(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure the proposal can be withdrawn.
+	if proposal.Status != group.ProposalStatusSubmitted {
+		return nil, sdkerrors.Wrapf(errors.ErrInvalid, "cannot withdraw a proposal with a status of %s", proposal.Status.String())
+	}
+
+	var policyInfo group.GroupPolicyInfo
+	if policyInfo, err = k.getGroupPolicyInfo(ctx, proposal.Address); err != nil {
+		return nil, sdkerrors.Wrap(err, "load group policy")
+	}
+
+	groupInfo, err := k.getGroupInfo(ctx, policyInfo.GroupId)
+	if err != nil {
+		return nil, err
+	}
+
+	if groupInfo.Version != proposal.GroupVersion {
+		return nil, sdkerrors.Wrap(errors.ErrModified, "group was already modified")
+	}
```

Comment:
why do we need to check for this? if the group had changed, then the proposal status would be `ProposalStatusAborted` so it would have been caught by line 499